pkg/chunked/internal/minimal/compression.go

```diff
@@ -244,7 +244,39 @@ func WriteZstdChunkedManifest(dest io.Writer, outMetadata map[string]string, off
 	return appendZstdSkippableFrame(dest, manifestDataLE)
 }
 
-func ZstdWriterWithLevel(dest io.Writer, level int) (*zstd.Encoder, error) {
+type ZstdWriter interface {
+	io.WriteCloser
+	Reset(dest io.Writer)
+	Flush() error
+}
+
+type CreateZstdWriterFunc func(dest io.Writer) (ZstdWriter, error)
+
+type noCompression struct {
+	dest io.Writer
+}
+
+func (n *noCompression) Write(p []byte) (int, error) {
+	return n.dest.Write(p)
+}
+
+func (n *noCompression) Close() error {
+	return nil
+}
+
+func (n *noCompression) Flush() error {
+	return nil
+}
+
+func (n *noCompression) Reset(dest io.Writer) {
+	n.dest = dest
+}
+
+func NoCompression(dest io.Writer) (ZstdWriter, error) {
```

Comment:
Could this be moved to `chunked/compressor`? `internal/minimal` is used even by `chunked/toc`, so including any more code here is not ideal, and there is already an interface . (OTOH, admittedly, for c/image, it would be included either way…)

… Actually can’t most of the writer code here be moved into `chunked/compressor` wholesale, leaving `minimal` to be mostly type definitions? The readers in `pkg/chunked`, separate from the writers, anyway.